tools/src/main/scala/scala/scalanative/build/Filter.scala

```diff
@@ -33,44 +45,83 @@ private[scalanative] object Filter {
   ): (Seq[Path], Config) = {
     val nativeCodePath = destPath.resolve(nativeCodeDir)
     // check if filtering is needed, o.w. return all paths
-    findFilterProperties(nativeCodePath).fold((allPaths, config)) { file =>
-      // predicate to check if given file path shall be compiled
-      // we only include sources to the base of the gc code and exclude
-      // all optional dependencies if they are not necessary
-      val optPath = nativeCodePath.resolve("optional").abs
-      val gcPath = nativeCodePath.resolve("gc").abs
-
-      def include(path: String) = {
-        if (path.contains(optPath)) {
-          val name = Paths.get(path).toFile.getName.split("\\.").head
-          linkerResult.links.exists(_.name == name)
-        } else {
-          true
+    findFilterProperties(nativeCodePath).fold((allPaths, config)) { filepath =>
+
+      val desc =
+        tl.Using(new FileReader(filepath.toFile())) { reader =>
```

Comment:
`Using` of course is a noice addition, but `try-catch-finally {reader.close()}` would the job as well, and would be easier to maintain

Reply:
Sounds good. Had to brush up on that, very Java like - hopefully I got it correct.